rmw_fastrtps_cpp/include/rmw_fastrtps_cpp/TypeSupport_impl.h

```diff
@@ -513,7 +575,18 @@ void deserialize_array<std::string>(
 {
     (void)call_new;
     if (member->array_size_ && !member->is_upper_bound_) {
-        deser.deserializeArray(((rosidl_generator_c__String*)field)->data, member->array_size_);
+        auto deser_field = (rosidl_generator_c__String*)field;
+        std::string stringseq[member->array_size_];
+        deser.deserializeArray(stringseq, member->array_size_);
+        for (size_t i = 0; i < member->array_size_; ++i) {
+          // NOTE: Do we need to call init ?
```

Comment:
After discussing this I think we have to assume that the passed in `deser_field[i]` has been correctly initialized before calling this. (We can't even check for that since uninitialized is not distinguishable.) It is also not coming from any user land code but our own rmw code. So if we don't screw it up there is will be initialized correctly. So the init call needs to be removed again. Sorry for the back and forth.